cogs/helpers.py

```diff
@@ -643,14 +645,23 @@ async def previous_roles(self, ctx, user: discord.Member):
             while p.edit_mode:
                 if discord.utils.get(ok_user.roles,
                                      name=self.bot.config.moderator_role):
-                    await user.add_roles(
-                        *valid_roles,
-                        reason="{} used the previous_roles command".format(
-                            ok_user.name))
+                    failed_roles: list[str] = []
+                    for role in valid_roles:
+                        try:
+                            await user.add_roles(
+                                role,
+                                reason=
+                                f"{ok_user.name} used the previous_roles command"
+                            )
+                        except:
```

Comment:
do you know the specific exception that occurs? its good to put the exception type here